index-handlers/src/main/java/no/sikt/nva/nvi/index/Aggregations.java

```diff
@@ -39,26 +28,20 @@
 public final class Aggregations {
 
     public static final int MULTIPLE = 2;
+    public static final String ALL_AGGREGATIONS = "all";
     private static final CharSequence JSON_PATH_DELIMITER = ".";
 
     private Aggregations() {
     }
 
-    public static Map<String, Aggregation> generateAggregations(String username, String customer) {
+    public static Map<String, Aggregation> generateAggregations(String aggregationType, String username,
+                                                                String topLevelCristinOrg) {
         var aggregations = new HashMap<String, Aggregation>();
-        aggregations.put(NEW_AGG.getAggregationName(), statusAggregation(customer, NEW));
-        aggregations.put(NEW_COLLABORATION_AGG.getAggregationName(), collaborationAggregation(customer, NEW));
-        aggregations.put(PENDING_AGG.getAggregationName(), statusAggregation(customer, PENDING));
-        aggregations.put(PENDING_COLLABORATION_AGG.getAggregationName(), collaborationAggregation(customer, PENDING));
-        aggregations.put(APPROVED_AGG.getAggregationName(), statusAggregation(customer, APPROVED));
-        aggregations.put(APPROVED_COLLABORATION_AGG.getAggregationName(),
-                         finalizedCollaborationAggregation(customer, APPROVED));
-        aggregations.put(REJECTED_AGG.getAggregationName(), statusAggregation(customer, REJECTED));
-        aggregations.put(REJECTED_COLLABORATION_AGG.getAggregationName(),
-                         finalizedCollaborationAggregation(customer, REJECTED));
-        aggregations.put(ASSIGNMENTS_AGG.getAggregationName(), assignmentsAggregation(username, customer));
-        aggregations.put(COMPLETED_AGGREGATION_AGG.getAggregationName(), completedAggregation(customer));
-        aggregations.put(TOTAL_COUNT_AGGREGATION_AGG.getAggregationName(), totalCountAggregation(customer));
+        if (aggregationTypeIsNotSpecified(aggregationType)) {
+            generateAllAggregationTypes(username, topLevelCristinOrg, aggregations);
+        } else {
+            generateSingleAggregation(aggregationType, username, topLevelCristinOrg, aggregations);
+        }
         return aggregations;
```

Comment:
It seems we don't need to initialize an empty map before the if-else conditional. Both generateAllAggregationTypes and generateSingleAggregation methods can directly return a new map.